reference/5.1/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md

```diff
@@ -619,11 +593,117 @@ The following example shows a parameter declaration that adds the **CN** and
 **MachineName** aliases to the mandatory **ComputerName** parameter.
 
 ```powershell
-Param(
+param(
     [Parameter(Mandatory)]
     [Alias("CN","MachineName")]
-    [string[]]
-    $ComputerName
+    [string[]]$ComputerName
+)
+```
+
+### Credential attribute
+
+The **Credential** attribute is used to indicate that the parameter accepts
+credentials. The following example shows a parameter declaration that uses the
+**Credential** attribute.
+
+```powershell
+param(
+    [Parameter()]
+    [System.Management.Automation.Credential()]
+    [PSCredential]$Credential
+)
+```
+
+### PSDefaultValue attribute
+
+The **PSDefaultValue** specifies the default value of a command parameter in a
+script. This information is displayed by the `Get-Help` cmdlet. To see the
+default value information, the function must include comment-based help. For
+example:
+
+```powershell
+<#
+    .SYNOPSIS
+     This is a test script that has a parameter with a default value.
+#>
+function TestDefaultValue {
+    param(
+        [PSDefaultValue(Help='Current directory')]
+        [string]$Name = $PWD.Path
+    )
+
+    $Name
+}
+```
+
+Use `Get-Help` to see the default value information.
+
+```powershell
+Get-Help TestDefaultValue -Parameter name
+```
+
+```Output
+-Name <String>
+
+    Required?                    false
+    Position?                    1
+    Default value                Current directory
+    Accept pipeline input?       false
+    Accept wildcard characters?  false
+```
+
+#### PSDefaultValue attribute arguments
+
+The **PSDefaultValue** attribute has two arguments:
+
+- **Help** - A string that describes the default value. This information is
+  displayed by the `Get-Help` cmdlet.
+- **Value** - The default value of the parameter.
+
+Both arguments are optional. If you don't specify any arguments, then
+`Get-Help` shows the value assigned to the parameter.
+
+### PSTypeName attribute
+
+You can't use extended type names in a type declaration. The *PSTypeName**
+attribute allows you to restrict the type of the parameter to the extended
+type.
+
+In this example, the `Test-Connection` cmdlet returns an extended type. You can
+use the **PSTypeName** attribute to restrict the type of the parameter to the
+extended type.
+
+```powershell
+function TestType {
+    param(
+        [PSTypeName('System.Management.ManagementObject#root\cimv2\Win32_PingStatus')]
+        [psobject]$PingStatus
+    )
+
+    $PingStatus
+}
+
+$ping = Test-Connection bing.com -Count 1
+TestType $ping
+```
+
+### System.Obsolete attribute
+
+Use the **System.Obsolete** attribute to mark parameters that are no longer in
+use. This can be useful when you want to remove a parameter from a function but
+you don't want to break existing scripts that use the function.
```

Comment:
```suggestion
Use the **System.Obsolete** attribute to mark parameters that are no longer
supported. This can be useful when you want to remove a parameter from a
function but you don't want to break existing scripts that use the function.
```